pkg/kubelet/server/stats/volume_stat_calculator.go

```diff
@@ -133,7 +134,11 @@ func (s *volumeStatCalculator) calcAndStoreStats() {
 	var ephemeralStats []stats.VolumeStats
 	var persistentStats []stats.VolumeStats
 	for name, v := range metricVolumes {
-		metric, err := v.GetMetrics()
+		metric, err := func() (*volume.Metrics, error) {
+			startTime := time.Now()
+			defer servermetrics.CollectVolumeStatCalDuration("all", startTime)
```

Comment:
You are right. This is not reasonable.

The `cached` and `etc-hosts`/`hostspath` are also not valid. 
- `etc-hosts`/`hostspath` are using `DuMetrics` and they will be included in the `du` volume type.
- `cached` is not a type, it will also call other metrics.
- `all` is from the pods aspect, but it will belong to `block` or `du` or other types.

I updated the types.
- du
- block
- csi (use driver-name as the volume type)
- statfs